backend/src/packages/articles/article.service.ts

```diff
@@ -342,6 +404,12 @@ class ArticleService implements IService {
       throw new ForbiddenError('Article can be edited only by author!');
     }
 
+    let updatedGenreId = payload.genreId ?? article.genreId;
+
+    if (payload.text && (article.promptId || payload.promptId)) {
```

Comment:
```suggestion
    const hasToCheckActualGenre = payload.text && (article.promptId || payload.promptId);
    if (hasToCheckActualGenre) {
```

Reply:
you will lose payload.text type guard this way and will have to check it again